### ISSUE TITLE
Turn off profile update request form

### ISSUE THREAD

**Issue by pbanaszkiewicz** [swcarpentry/amy#865]
Since we allowed instructors to log in and update their profiles via `AutoUpdateForm`, we probably don't need the external (open for not-logged in users) form `ProfileUpdateRequest`.

@gvwilson needs to remind instructors to add their GitHub link because some cannot log in yet.

@maneesha please confirm you won't need the `ProfileUpdateRequest` form ( https://amy.software-carpentry.org/workshops/update_profile/ )

**Comment by maneesha** [comment]
I  go in once a week or so to process all the Profile Update Requests that have come in.  I don't use it for anything else.  However if Github login is required, and Data Carpentry does not require instructors to be Github users this form may still be necessary.  @tracykteal?

**Comment by pbanaszkiewicz** [comment]
@tracykteal can you please update us on this matter?

**Comment by tracykteal** [comment]
Since DC doesn't require that people know github or check out on it for instructor training, and therefore people don't necessarily have github accounts, it would be best if we could still allow logins that are not github usernames. Is that possible?

**Comment by gvwilson** [comment]
Managing passwords and password updates in Django is a pain, so we'd strongly prefer to delegate to external services.  @pbanaszkiewicz what other authentication services can we enable? And @tracykteal do you have favorites?

**Comment by tracykteal** [comment]
I would say that Google is the most widely used.

**Comment by pbanaszkiewicz** [comment]
@chrismedrela we can have Google login pretty quickly now, can we?

**Comment by chrismedrela** [comment]
I had a look at documentation and there is nothing catchy. My only concern is how do we associate Google account with AMY account? What we do for GitHub now is that we know person's GitHub username and, given this fact, we manually synchronize `UserSocialAuth` table. Do we want similar workflow for Google auth?

**Comment by gvwilson** [comment]
We can ask people to give us their Google ID in their instructor profile 
if they want to be able to log in that way.

**Comment by tracykteal** [comment]
Thanks for looking in to this. Having Google ID in the instructor profile seems like it would work. Then could people log in with either Google or github if they have both in their profile? Or will they need to remember which they used?

**Comment by chrismedrela** [comment]
If they give us both GitHub username and Google ID, they can log in with either Google or GitHub.

**Comment by tracykteal** [comment]
That sounds great. Thanks.

**Comment by pbanaszkiewicz** [comment]
Isn't Google ID one's email?

**Comment by tracykteal** [comment]
Yes, if it's your Google email. I'm tkteal@gmail.com, so my Google ID is 'tkteal'

**Comment by gvwilson** [comment]
@pbanaszkiewicz @chrismedrela have we tested login with Google ID? If so, and if it works, I think this one's done.

**Comment by chrismedrela** [comment]
@gvwilson [we don't have google login implemented](https://github.com/swcarpentry/amy/blob/develop/amy/settings.py#L218-L221), only GitHub and standard login+password way.

**Comment by gvwilson** [comment]
How much work to enable Google login?

**Comment by chrismedrela** [comment]
IMO 8 hours.

[`python-social-auth` part seems to be easy](http://psa.matiasaguirre.net/docs/backends/google.html#google-oauth2), but we need to think out how we keep `UserSocialAuth` and `Person` in sync (we encountered similar issue with `UserSocialAuth` and `Person.github`). The easiest way is to modify `Person.check_if_usersocialauth_is_in_sync` and `Person.synchronize_usersocialauth`, so that it takes Google auth into account too. @pbanaszkiewicz any opinions on implementation?

**Comment by gvwilson** [comment]
OK, let's make this high priority for the December release.

**Comment by pbanaszkiewicz** [comment]
> OK, let's make this high priority for the December release.

Then this issue depends on #1049, therefore I'm moving it to v1.10.

> @pbanaszkiewicz any opinions on implementation?

I still believe we should make use of `UserSocialAuth` table, drop magic surrounding `Person.github`, and implement a proper "Manage your social auth links" page for people.

**Comment by chrismedrela** [comment]
> I still believe we should make use of UserSocialAuth table, drop magic surrounding Person.github, and implement a proper "Manage your social auth links" page for people.

How do we let trainees log in for the first time when we don't have their GitHub/Google account associated and they don't have login+password pair?

**Comment by pbanaszkiewicz** [comment]
>How do we let trainees log in for the first time when we don't have their GitHub/Google account associated and they don't have login+password pair?

I cannot think of any other solution than an invitation for registration system.

**Comment by chrismedrela** [comment]
This depends on #1049, which I postponed to v1.11. Therefore, I'm postponing this issue too.

**Comment by pbanaszkiewicz** [comment]
Closing in favor of #1371.
